src/code/z_actor.c

```diff
@@ -457,439 +1246,3471 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B882C.s")
+    if (flag != 0) {
+        Gfx* displayListHead;
+        Gfx* displayList = GRAPH_ALLOC(globalCtx->state.gfxCtx, 2 * sizeof(Gfx));
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B886C.s")
+        displayListHead = displayList;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8898.s")
+        OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8934.s")
+        gDPSetHilite1Tile(displayListHead++, 1, hilite, 0x10, 0x10);
+        gSPEndDisplayList(displayListHead);
+        gSPSegment(POLY_XLU_DISP++, 0x07, displayList);
 
-u32 Actor_HasParent(Actor* actor, GlobalContext* globalCtx) {
-    if (actor->parent != NULL) {
-        return true;
-    } else {
-        return false;
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8A1C.s")
+PosRot* Actor_GetFocus(PosRot* dest, Actor* actor) {
+    *dest = actor->focus;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8B84.s")
+    return dest;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8BB0.s")
+PosRot* Actor_GetWorld(PosRot* dest, Actor* actor) {
+    *dest = actor->world;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8BD0.s")
+    return dest;
+}
 
-s32 Actor_HasNoParent(Actor* actor, GlobalContext* globalCtx) {
-    if (!actor->parent) {
-        return true;
-    } else {
-        return false;
+PosRot* Actor_GetWorldPosShapeRot(PosRot* dest, Actor* actor) {
+    PosRot sp1C;
+
+    Math_Vec3f_Copy(&sp1C.pos, &actor->world.pos);
+    if (actor->id == ACTOR_PLAYER) {
+        Player* player = (Player*)actor;
+
+        sp1C.pos.y += player->unk_AC0 * actor->scale.y;
     }
+    sp1C.rot = actor->shape.rot;
+    *dest = sp1C;
+
+    return dest;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8C20.s")
+f32 func_800B82EC(Actor* actor, Player* player, s16 angle) {
+    f32 temp_f12;
+    s16 temp_v0 = BINANG_SUB(BINANG_SUB(actor->yawTowardsPlayer, 0x8000), angle);
+    s16 yaw = ABS_ALT(temp_v0);
+
+    if (player->unk_730 != NULL) {
+        if ((yaw > 0x4000) || ((actor->flags & ACTOR_FLAG_8000000))) {
+            return FLT_MAX;
+        }
+
+        temp_f12 = actor->xyzDistToPlayerSq - ((actor->xyzDistToPlayerSq * 0.8f) * ((0x4000 - yaw) * (1.0f / 0x8000)));
+        return temp_f12;
+    }
+
+    if (yaw >= 0x2AAB) {
+        return FLT_MAX;
+    }
+    return actor->xyzDistToPlayerSq;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_actor/func_800B8C50.s")
+TargetRangeParams D_801AECF0[] = {
+    TARGET_RANGE(70, 140),   TARGET_RANGE(170, 255),    TARGET_RANGE(280, 5600),      TARGET_RANGE(350, 525),
+    TARGET_RANGE(700, 1050), TARGET_RANGE(1000, 1500),  TARGET_RANGE(100, 105.36842), TARGET_RANGE(140, 163.33333),
+    TARGET_RANGE(240, 576),  TARGET_RANGE(280, 280000), TARGET_RANGE(2500, 3750),
+};
```

Comment:
Could you explain what this data is doing if you know? This data is actually directly accessed in camera and used in a way I don't understand. Any way you could provide insight into this function below?
```
// TODO: D_801AECF0 as Vec2f matched, but real data type is unknown
// Note: D_801AECF0 size is unknown but OoT arrays indexed by actor->targetMode are size 10
f32 func_800CD6CC(Actor* actor) {
    return sqrtf(D_801AECF0[actor->targetMode].x / D_801AECF0[actor->targetMode].y);
}
```

Reply:
Sorry, I'm not really sure why that function is doing that.

I don't know if you have checked, but I have that struct and that macro defined as this:
```c
typedef struct {
    /* 0x0 */ f32 rangeSq;
    /* 0x4 */ f32 leashScale;
} TargetRangeParams; // size = 0x8


#define TARGET_RANGE(range, leash) \
    { SQ(range), (f32)range / leash }

```
Which I took from OoT